CMakeLists.txt

```diff
@@ -1,504 +1,2030 @@
+  BYPRODUCTS ${CMAKE_BINARY_DIR}/sqlite3/include/sqlite3.h)
+
+#[===================================================================[
+   NIH dep: soci
+#]===================================================================]
+
+add_library (soci STATIC
+  #[=========================================================[
+     this unity file is our interpretation of
+     what sources comprise soci-lib.
+     (this must be investigated/confirmed whenever updated)
+  #]=========================================================]
+  src/ripple/unity/soci.cpp
+  # this header is here to create dependency on the sqlite post-build step
+  ${CMAKE_BINARY_DIR}/sqlite3/include/sqlite3.h)
+target_compile_definitions (soci PUBLIC SOCI_HAVE_CXX_C11=1)
+target_include_directories (soci
+  PUBLIC
+    src/soci/src
+    src/soci/include
+    #[=========================================================[
+       HACK for ninja..which doesn't properly see
+       the POST_BUILD dependency on sqlite (TODO: diagnose)
+       so we must include the original source dir
+    #]=========================================================]
+    $<$<STREQUAL:${CMAKE_GENERATOR},Ninja>:
+      ${CMAKE_CURRENT_SOURCE_DIR}/src/sqlite
+    >
+  PRIVATE
+    src
+    src/soci/src/core
+    src/soci/include/private)
+target_link_libraries (soci
+  NIH::sqlite3
+  Ripple::boost)
+add_library (NIH::soci ALIAS soci)
+target_link_libraries (ripple_libs INTERFACE NIH::soci)
+
+#[===================================================================[
+   NIH dep: snappy
+#]===================================================================]
+
+add_library (snappy STATIC
+  src/snappy/snappy/snappy.cc
+  src/snappy/snappy/snappy-sinksource.cc
+  src/snappy/snappy/snappy-stubs-internal.cc)
+target_compile_options (snappy
+  PRIVATE
+    $<$<NOT:$<BOOL:${MSVC}>>:
+      -Wno-deprecated
+      -Wno-unused-function
+    >
+    $<$<BOOL:${is_gcc}>:-Wno-nonnull-compare>)
+target_include_directories (snappy
+  PUBLIC
+    src/snappy/snappy
+    src/snappy/config)
+add_library (NIH::snappy ALIAS snappy)
+target_link_libraries (ripple_libs INTERFACE NIH::snappy)
+
+#[===================================================================[
+   NIH dep: rocksdb
+#]===================================================================]
+
+include (CheckCXXCompilerFlag)
+CHECK_CXX_COMPILER_FLAG ("-std=c++11" COMPILER_SUPPORTS_CXX11)
+if (NOT MSVC AND COMPILER_SUPPORTS_CXX11)
+  add_library (rocksdb STATIC
+    #[=========================================================[
+       this unity file is our interpretation of what sources
+       comprise rocksdb (this must be investigated/confirmed
+       whenever updated)
+    #]=========================================================]
+    src/ripple/unity/rocksdb.cpp)
+  target_compile_options (rocksdb
+    PRIVATE
+      $<$<BOOL:${is_gcc}>:-w>
+    PUBLIC
+      $<$<AND:$<BOOL:${is_clang}>,$<VERSION_GREATER_EQUAL:CMAKE_CXX_COMPILER_VERSION,7>>:
```

Comment:
Same comment as earlier as I think this is imposing `AppleClang` versioning numbers which won't map well to LLVM version numbers. For example, I compile using LLVM Clang 6.

Reply:
good point - I'll change this to a strict Clang compiler check